source/common/http/conn_manager_impl.cc

```diff
@@ -896,6 +929,11 @@ void ConnectionManagerImpl::ActiveStream::decodeHeaders(RequestHeaderMapPtr&& he
     request_header_timer_.reset();
   }
 
+  if (!validateHeaders()) {
+    ENVOY_STREAM_LOG(debug, "request headers validation failed:\n{}", *this, *request_headers_);
```

Comment:
encouraged follow-up (TODO?) we could have a default header validator, and move most of Envoy's current HCM checks there since they should be redundant with UHV
if (!request_headers_->Host())
  if ((!HeaderUtility::isConnect(*request_headers_) || request_headers_->Path()) &&
...

Reply:
This is on my TODO list and PR with these changes will be out soon.